src/logic/community-voice/validation.ts

```diff
@@ -0,0 +1,39 @@
+import { CommunityRole } from '../../types/entities'
+import { UserNotCommunityMemberError } from './errors'
+import { ICommunitiesDatabaseComponent } from '../../types/components'
+import { Community } from '../community/types'
+
+/**
+ * Helper function to validate target user permissions for voice chat operations (promote/demote) based on community privacy
+ * @param communitiesDb - Communities database adapter
+ * @param community - Community object with privacy information
+ * @param communityId - Community ID
+ * @param targetUserAddress - Target user address
+ * @returns Promise<void> - Throws error if validation fails
+ */
+export async function validateCommunityVoiceChatTargetUser(
+  communitiesDb: ICommunitiesDatabaseComponent,
+  community: Community,
+  communityId: string,
+  targetUserAddress: string
+): Promise<void> {
+  // Validation logic based on community type
```

Comment:
Can we do an early return if the community is not private to avoid the if/else? I think it could simplify the function